substrate/frame/support/procedural/src/construct_runtime/expand/inherent.rs

```diff
@@ -149,34 +150,31 @@ pub fn expand_outer_inherent(
 						}
 					)*
 
-					// Inherents are before any other extrinsics.
-					// No module marked it as inherent thus it is not.
-					if !is_inherent {
-						break
+
+					// If there is no pallet apart from System, the true branch will be unreachable.
+					#[allow(unreachable_code)]
+					match is_inherent {
+						true => {
+							inherents.push(call);
```

Comment:
Hope we won't ever have any inherents with `Xcm<()>` arguments :)

Reply:
Yes. I don't know if this can happen. Also `Xcm<()>` variables can be ok as long as they are not very big.